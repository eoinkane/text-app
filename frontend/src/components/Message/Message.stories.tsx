import React from "react";
import MessageComponent from "./index";
import Container from "@material-ui/core/Container";

export default {
  title: "Message",
  component: MessageComponent,
};

export const Message = () => {
  const text =
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ";
  return (
    <Container style={{ backgroundColor: "#f5f5f5", padding: "4vh 4vw" }}>
      <MessageComponent text={text} />
    </Container>
  );
};
