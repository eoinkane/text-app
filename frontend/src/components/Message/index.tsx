import React from "react";
import cx from "clsx";
import { Paper, Typography } from "@material-ui/core";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";

interface MessagePropsBase {}

interface MessagePropsLeft extends MessagePropsBase {
  left: boolean;
}

interface MessagePropsRight extends MessagePropsBase {
  right: boolean;
}

type MessageProps = MessagePropsLeft | MessagePropsRight;

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      maxWidth: "85%",
      padding: "10px 0",
    },
    left: {
      marginRight: "auto",
    },
    right: {
      marginLeft: "auto",
    },
    paper: {
      padding: "2vh 2vh",
      borderRadius: 10,
    },
  })
);

const Message: React.FC<MessageProps> = ({ children, ...rest }) => {
  const classes = useStyles();
  return (
    <div
      className={cx(
        classes.root,
        {
          [classes.left]:
            rest.hasOwnProperty("left") && (rest as MessagePropsLeft).left,
        },
        {
          [classes.right]:
            rest.hasOwnProperty("right") && (rest as MessagePropsRight).right,
        }
      )}
    >
      <Paper className={classes.paper} elevation={3}>
        <Typography>{children}</Typography>
      </Paper>
    </div>
  );
};

export default Message;
