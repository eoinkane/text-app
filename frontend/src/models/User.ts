import axios, { AxiosResponse } from "axios";

interface UserProps {
  firstName?: string;
  lastName?: string;
  id?: number;
}

type valueof<T> = T[keyof T];

export default class User {
  constructor(private data: UserProps) {}

  get<K extends keyof UserProps>(propName: K): valueof<UserProps> {
    return this.data[propName];
  }

  set(update: UserProps): void {
    Object.assign(this.data, update);
  }
  fetch(): void {
    // axios
    //   .get(`http://localhost:3000/users/${this.get("id")}`)
    //   .then((response: AxiosResponse): void => {
    //     this.set(response.data);
    //   });
  }

  save(): void {
    // const id = this.get("id");
    // if (id) {
    //   axios.put(`http://localhost:3000/users/${id}`, this.data);
    // } else {
    //   axios.post(`http://localhost:3000/users`, this.data);
    // }
  }
}
