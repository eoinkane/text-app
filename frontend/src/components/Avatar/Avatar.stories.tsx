import React from "react";
import User from "../../models/User/";
import AvatarComponent from "./index";

export default {
  title: "Avatar",
  component: AvatarComponent,
};

export const AvatarWithFullUser = () => {
  const user = new User();
  user.init({
    id: 0,
    username: "test",
    firstName: "Jane",
    lastName: "Doe",
  });
  return <AvatarComponent user={user} />;
};
