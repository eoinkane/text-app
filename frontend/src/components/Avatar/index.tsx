import React, { Fragment } from "react";
import User from "../../models/User/";
import { Avatar as MaterialUIAvatar } from "@material-ui/core";

interface IAvatarProps {
  user: User;
}

const Avatar: React.FC<IAvatarProps> = ({ user }) => {
  const firstName = user.get("firstName") as string;
  const lastName = user.get("lastName") as string;
  const initals =
    !firstName || !lastName ? null : `${firstName[0]}${lastName[0]}`;
  return (
    <Fragment>
      <MaterialUIAvatar>{initals}</MaterialUIAvatar>
    </Fragment>
  );
};

export default Avatar;
