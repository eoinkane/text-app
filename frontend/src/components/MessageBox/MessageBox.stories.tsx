import React from "react";
import MessageBoxComponent from "./index";
import User from "../../models/User";
import Message from "../../models/Message";
import { Container, Typography } from "@material-ui/core";
import { CurrentUserContext } from "../../contexts/CurrentUserContext";

export default {
  title: "MessageBox",
  component: MessageBoxComponent,
};

const sender = new User();
sender.init({
  username: "test",
  id: 0,
  firstName: "Alex",
  lastName: "Doe",
});

const currentUser = new User();
currentUser.init({
  username: "test_user",
  id: 0,
  firstName: "test",
  lastName: "user",
});

const message = new Message({
  message:
    "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
});

export const MessageBox = () => {
  return (
    <Container style={{ padding: "4vh 4vw" }}>
      <Typography style={{ textAlign: "center", paddingBottom: 10 }}>
        Recieved Message
      </Typography>
      <Container
        style={{
          backgroundColor: "#f5f5f5",
          margin: "20px 0",
        }}
      >
        <CurrentUserContext.Provider
          value={{
            currentUser: currentUser,
            initUser: () => {
              return Promise.resolve();
            },
            error: false,
          }}
        >
          <MessageBoxComponent sender={sender} message={message} />
        </CurrentUserContext.Provider>
      </Container>

      <Typography style={{ textAlign: "center", paddingBottom: 10 }}>
        Sent Aligned Message
      </Typography>
      <Container style={{ backgroundColor: "#f5f5f5", margin: "20px 0" }}>
        <CurrentUserContext.Provider
          value={{
            currentUser: currentUser,
            initUser: () => {
              return Promise.resolve();
            },
            error: false,
          }}
        >
          <MessageBoxComponent sender={currentUser} message={message} />
        </CurrentUserContext.Provider>
      </Container>
    </Container>
  );
};
