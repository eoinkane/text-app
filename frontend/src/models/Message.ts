import axios, { AxiosResponse } from "axios";
import User from "./User";

interface MessageProps {
  id?: number;
  sender?: User;
  message?: string;
  conversationsId?: number;
}

type valueof<T> = T[keyof T];

export default class Message {
  constructor(private data: MessageProps) {}

  get<K extends keyof MessageProps>(propName: K): valueof<MessageProps> {
    return this.data[propName];
  }

  set(update: MessageProps): void {
    Object.assign(this.data, update);
  }
  fetch(): void {
    // axios
    //   .get(`http://localhost:3000/users/${this.get("id")}`)
    //   .then((response: AxiosResponse): void => {
    //     this.set(response.data);
    //   });
  }

  save(): void {
    // const id = this.get("id");
    // if (id) {
    //   axios.put(`http://localhost:3000/users/${id}`, this.data);
    // } else {
    //   axios.post(`http://localhost:3000/users`, this.data);
    // }
  }
}
