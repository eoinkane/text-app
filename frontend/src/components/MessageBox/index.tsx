import React from "react";
import Avatar from "./../Avatar";
import MessageComponent from "./../Message";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import User from "../../models/User";
import Message from "../../models/Message";
import { Grid } from "@material-ui/core";

interface MessageBoxProps {
  sender: User;
  currentUser: User;
  message: Message;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      flexGrow: 1,
      padding: "10px 0",
    },
    rightAvatarGrid: {
      flexBasis: 0,
    },
  })
);

const MessageBox: React.FC<MessageBoxProps> = ({
  sender,
  currentUser,
  message,
}) => {
  const classes = useStyles();
  const messageSentByCurrentUser = sender === currentUser;
  return (
    <div className={classes.root}>
      <Grid
        container
        direction="row"
        justify="flex-start"
        alignItems="flex-end"
      >
        {!messageSentByCurrentUser ? (
          <Grid item xs={1}>
            <Avatar user={sender} />
          </Grid>
        ) : null}
        <Grid item xs={11}>
          <MessageComponent
            left={!messageSentByCurrentUser}
            right={messageSentByCurrentUser}
          >
            {message.get("message") as string}
          </MessageComponent>
        </Grid>
        {messageSentByCurrentUser ? (
          <Grid className={classes.rightAvatarGrid} item xs={1}>
            <Avatar user={sender} />
          </Grid>
        ) : null}
      </Grid>
    </div>
  );
};

export default MessageBox;
