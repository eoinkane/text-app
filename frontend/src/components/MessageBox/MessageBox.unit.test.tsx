import React from "react";
import { shallow } from "enzyme";
import MessageBox from "./index";
import User from "../../models/User";
import Message from "../../models/Message";
import MessageComponent from "../Message";
import { Grid } from "@material-ui/core";
import Avatar from "../Avatar";

describe("MessageBox", () => {
  const sender = new User({ firstName: "Jane", lastName: "Doe" });
  const currentUser = new User({ firstName: "Alex", lastName: "Doe" });
  const message = new Message({
    message:
      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. ",
  });

  it("Renders ", () => {
    const messageBox = shallow(
      <MessageBox sender={sender} currentUser={currentUser} message={message} />
    );
    expect(messageBox).toMatchSnapshot();
  });

  it("Renders the correct text", () => {
    const messageBox = shallow(
      <MessageBox sender={sender} currentUser={currentUser} message={message} />
    );
    expect(messageBox).toMatchSnapshot();

    expect(messageBox.find(MessageComponent).dive().text()).toEqual(
      message.get("message") as string
    );
  });

  it("Renders Avater on right for sent messages", () => {
    const messageBox = shallow(
      <MessageBox sender={sender} currentUser={sender} message={message} />
    );
    expect(messageBox).toMatchSnapshot();

    expect(messageBox.find(Grid).last().find(Avatar).length).toBe(1);
  });

  it("Renders Avater on left for recieved messages", () => {
    const messageBox = shallow(
      <MessageBox sender={sender} currentUser={currentUser} message={message} />
    );
    expect(messageBox).toMatchSnapshot();

    expect(
      messageBox.find(Grid).first().dive().find(Grid).find(Avatar).length
    ).toBe(1);
  });

  it("Renders the correct padding for right avatars", () => {
    const messageBox = shallow(
      <MessageBox sender={sender} currentUser={sender} message={message} />
    );
    expect(messageBox).toMatchSnapshot();

    expect(
      messageBox.find(Grid).last().hasClass("makeStyles-rightAvatarGrid-2")
    ).toBe(true);
  });
});
