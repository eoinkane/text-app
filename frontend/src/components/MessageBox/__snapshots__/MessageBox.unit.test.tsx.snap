// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox Renders  1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    container={true}
    direction="row"
    justify="flex-start"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={1}
    >
      <Avatar
        user={
          User {
            "data": Object {
              "firstName": "Jane",
              "lastName": "Doe",
            },
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={11}
    >
      <Message
        left={true}
        right={false}
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
      </Message>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`MessageBox Renders Avater on left for recieved messages 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    container={true}
    direction="row"
    justify="flex-start"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={1}
    >
      <Avatar
        user={
          User {
            "data": Object {
              "firstName": "Jane",
              "lastName": "Doe",
            },
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={11}
    >
      <Message
        left={true}
        right={false}
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
      </Message>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`MessageBox Renders Avater on right for sent messages 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    container={true}
    direction="row"
    justify="flex-start"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={11}
    >
      <Message
        left={false}
        right={true}
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
      </Message>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-rightAvatarGrid-2"
      item={true}
      xs={1}
    >
      <Avatar
        user={
          User {
            "data": Object {
              "firstName": "Jane",
              "lastName": "Doe",
            },
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`MessageBox Renders the correct padding for right avatars 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    container={true}
    direction="row"
    justify="flex-start"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={11}
    >
      <Message
        left={false}
        right={true}
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
      </Message>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-rightAvatarGrid-2"
      item={true}
      xs={1}
    >
      <Avatar
        user={
          User {
            "data": Object {
              "firstName": "Jane",
              "lastName": "Doe",
            },
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;

exports[`MessageBox Renders the correct text 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Grid))
    alignItems="flex-end"
    container={true}
    direction="row"
    justify="flex-start"
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={1}
    >
      <Avatar
        user={
          User {
            "data": Object {
              "firstName": "Jane",
              "lastName": "Doe",
            },
          }
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={11}
    >
      <Message
        left={true}
        right={false}
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ut eu consequat dui. Vivamus arcu tellus, tristique at purus ut, sagittis volutpat justo. Phasellus imperdiet ex. 
      </Message>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
